From 74cfb2f28b74908e44b467eb83049b337a45037d Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Fri, 15 Jun 2018 22:22:22 +0000
Subject: [PATCH] TRITON-504 NAPI unit tests broken by TRITON-474 Reviewed by:
 Mike Zeller <mike.zeller@joyent.com> Approved by: Mike Zeller
 <mike.zeller@joyent.com>

---
 lib/napi.js      | 21 ++++++++++++++-------
 test/config.json |  4 ++++
 2 files changed, 18 insertions(+), 7 deletions(-)

diff --git a/lib/napi.js b/lib/napi.js
index 465a301..6d1147b 100644
--- a/lib/napi.js
+++ b/lib/napi.js
@@ -94,7 +94,7 @@ function NAPI(opts) {
     http.globalAgent.maxSockets = maxSockets;
     https.globalAgent.maxSockets = maxSockets;
 
-    var metricsManager = createMetricsManager({
+    this.metricsManager = createMetricsManager({
         address: opts.config.adminIp,
         log: opts.log.child({component: 'metrics'}),
         port: METRICS_SERVER_PORT,
@@ -107,8 +107,8 @@ function NAPI(opts) {
         }
     });
 
-    metricsManager.createRestifyMetrics();
-    metricsManager.listen(function metricsServerStarted() {});
+    this.metricsManager.createRestifyMetrics();
+    this.metricsManager.listen(function metricsServerStarted() {});
 
     function populateReq(req, res, next) {
         req.config = opts.config;
@@ -197,8 +197,9 @@ function NAPI(opts) {
         })(req, res, route, err);
     });
 
-    server.on('after', metricsManager.collectRestifyMetrics
-        .bind(metricsManager));
+    server.on('after', function (req, res, route) {
+        self.metricsManager.collectRestifyMetrics(req, res, route);
+    });
 
     endpoints.registerEndpoints(server, before, this.log);
 
@@ -405,7 +406,11 @@ NAPI.prototype.state_stopped = function (S) {
 NAPI.prototype._cleanup = function (callback) {
     var self = this;
 
-    this.server.close(function (err) {
+    function onMetricsClose() {
+        self.server.close(onServerClose);
+    }
+
+    function onServerClose(err) {
         if (self.moray) {
             self.moray.close();
         }
@@ -418,7 +423,9 @@ NAPI.prototype._cleanup = function (callback) {
         if (callback) {
             callback(err);
         }
-    });
+    }
+
+    this.metricsManager.close(onMetricsClose);
 };
 
 
diff --git a/test/config.json b/test/config.json
index d5b43dc..e8d4ada 100644
--- a/test/config.json
+++ b/test/config.json
@@ -3,6 +3,10 @@
 
   "allowLinklocal": true,
   "datacenter": "unit-test",
+  "serviceName": "napi",
+  "instanceUuid": "09f64a09-7bef-c977-91cf-dd25fa5474e8",
+  "serverUuid": "09f64a09-7bef-c977-91cf-dd25fa5474e8",
+  "adminIp": "127.0.0.1",
   "autoAllocSubnets": false,
   "macOUI": "010203",
   "overlay": {
-- 
2.21.0

